docs/DeveloperGuide.md

```diff
@@ -278,142 +276,109 @@ Below is a class diagram describing the implementation of `Task` and its respect
+Student to a Consultation.
 
-**Note:** If the `currentStatePointer` is at index 0, pointing to the initial AddressBook state, then there are no previous AddressBook states to restore. The `undo` command uses `Model#canUndoAddressBook()` to check if this is the case. If so, it will return an error to the user rather
-than attempting to perform the undo.
+#### The Consultation Class
 
-</box>
+The Consultation Class is made up of a `LocalDate`, `LocalTime`, a `HashSet` of `Person` and a set of getter methods
+that corresponds to these fields.
 
-The following sequence diagram shows how the undo operation works:
+Below is a class diagram describing the implementation of `Consultation` and its respective fields.
 
-<puml src="diagrams/UndoSequenceDiagram.puml" alt="UndoSequenceDiagram" />
+<p align="center"><img src="images/ConsultationClassUMLDiagram.png"></p>
+<p align="center">Consultation Class UML Diagram</p>
 
-<box type="info" seamless>
+#### Design Considerations:
 
-**Note:** The lifeline for `UndoCommand` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+**Aspect: How the students are stored to a consultation:**
 
-</box>
+* **Alternative 1 (Current choice):** Use Set<Person> to keep track of students in a consultation.
+    * Pros: Store only one of the same person, no duplicates.
+    * Cons: May have performance issues in terms of memory usage.
 
-The `redo` command does the opposite — it calls `Model#redoAddressBook()`, which shifts the `currentStatePointer` once to the right, pointing to the previously undone state, and restores the address book to that state.
 
-<box type="info" seamless>
+* **Alternative 2:** Use ArrayList<Person> to keep track of students.
+    * Cons: We must ensure there are no duplicates with additional checks.
 
-**Note:** If the `currentStatePointer` is at index `addressBookStateList.size() - 1`, pointing to the latest address book state, then there are no undone AddressBook states to restore. The `redo` command uses `Model#canRedoAddressBook()` to check if this is the case. If so, it will return an error to the user rather than attempting to perform the redo.
+**Aspect: Adding students to a new or existing consultation:**
 
-</box>
+* **Alternative 1 (Current choice):** The `AddToConsult` feature creates a new Consultation object with updated student list
+  * Pros: Defensive programming when entirely creating a new Consultation object without modifying previous object.
+  * Cons: Require additional checking to inform exception cases.
 
-Step 5. The user then decides to execute the command `list`. Commands that do not modify the address book, such as `list`, will usually not call `Model#commitAddressBook()`, `Model#undoAddressBook()` or `Model#redoAddressBook()`. Thus, the `addressBookStateList` remains unchanged.
+* **Alternative 2:** `AddToConsult` directly manipulate the attribute students in a Consultation object.
```

Comment:
Might want to use markdowns here: attribute `students`

Reply:
I agree